dask/array/tests/test_atop.py

```diff
@@ -324,7 +324,50 @@ def test_optimize_blockwise():
     )
 
 
-def test_optimize_blockwise_annotations():
+def test_optimize_blockwise_control_annotations():
+    """
+    Can we fuse blockwise layers with different, but compatible
+    annotations for retries, priority, etc.
+    """
+
+    a = da.ones(10, chunks=(5,))
+    b = a + 1
+
+    with dask.annotate(retries=5, workers=["a", "b", "c"], allow_other_workers=False):
+        c = b + 2
+
+    with dask.annotate(priority=2, workers=["b", "c", "d"], allow_other_workers=True):
+        d = c + 3
+
+    with dask.annotate(retries=3, resources={"GPU": 2, "Memory": 10}):
+        e = d + 4
+
+    with dask.annotate(priority=4, resources={"GPU": 5, "Memory": 4}):
+        f = e + 5
+
+    g = f + 6
+
+    dsk = da.optimization.optimize_blockwise(g.dask)
+
+    # Layers and their annotations should be fusable
```

Comment:
Can we also include a simple check that adding a custom annotation to one of the layers will disable fusion? For example:

```python
with dask.annotate(foo="bar"):
    h = g + 7
    
assert len(da.optimization.optimize_blockwise(h.dask).layers) == 2
```

Reply:
Good idea.